src/util/time.h

```diff
@@ -14,6 +14,16 @@
 
 using namespace std::chrono_literals;
 
+/** Mockable clock in the context of tests, otherwise the system clock */
+struct NodeClock : std::chrono::system_clock {
```

Comment:
Public in structs might be unneeded, as access is public by default in structs / private by default in classes, and because https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c8-use-class-rather-than-struct-if-any-member-is-non-public

Reply:
Yeah, it's definitely unneeded, just seems clearer for people to have it there explicitly? Seems like we usually have an explicit `public` when inheriting a `struct` from some other `struct`.

Reply:
Yes, good point re inheriting.

Reply:
Thx, added `public` inheritance